### PR TITLE
Update SDK to 8.0.100-rc.1.23381.2

### PR DESCRIPTION
**Pull request by eerhardt**
Testing for https://github.com/dotnet/runtime/issues/89734. Trying to see if we can repro the issue.

### CODE REVIEW

**Comment by eerhardt** [comment]
@lewing @radical @javiercn - Looks like everything is failing with 

```
/Users/runner/work/1/s/.dotnet/sdk/8.0.100-rc.1.23381.2/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To build this project, the following workloads must be installed: wasm-tools [/Users/runner/work/1/s/src/Components/WebAssembly/testassets/WasmLinkerTest/WasmLinkerTest.csproj]
/Users/runner/work/1/s/.dotnet/sdk/8.0.100-rc.1.23381.2/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To install these workloads, run the following command: dotnet workload restore [/Users/runner/work/1/s/src/Components/WebAssembly/testassets/WasmLinkerTest/WasmLinkerTest.csproj]
```

Can someone fix it?

**Comment by javiercn** [comment]
@eerhardt did this fail on retry?

**Comment by lewing** [comment]
@radical can you take a look?

File: src/Tools/Shared/CommandLine/CommandLineApplicationExtensions.cs [hunk 1]
```diff
@@ -26,7 +26,7 @@ public static CommandOption Option(this CommandLineApplication command, string t
         => command.Option(
             template,
             description,
-            template.IndexOf("<", StringComparison.Ordinal) != -1
+            template.Contains("<")
```
Comment (BrennanConroy) [inline src/Tools/Shared/CommandLine/CommandLineApplicationExtensions.cs:29]:
```suggestion
            template.Contains('<')
```

Comment (eerhardt) [reply src/Tools/Shared/CommandLine/CommandLineApplicationExtensions.cs]: This just re-broke the build... :( I'm reverting.

Comment (BrennanConroy) [reply src/Tools/Shared/CommandLine/CommandLineApplicationExtensions.cs:29]: Reverting won't fix it either. This must be used in multiple tfms, one without char based `Contains` method.

Comment (eerhardt) [reply src/Tools/Shared/CommandLine/CommandLineApplicationExtensions.cs:29]: Reverting your change will fix it.

Comment (eerhardt) [reply src/Tools/Shared/CommandLine/CommandLineApplicationExtensions.cs:29]: This set of analysis rules is terrible. Am I really supposed to #if here?

Comment (eerhardt) [reply src/Tools/Shared/CommandLine/CommandLineApplicationExtensions.cs]: This is a bad experience.

**Review by sebastienros — APPROVED** [review]

**Pull request merged**